src/java/htsjdk/samtools/Defaults.java

```diff
@@ -73,7 +82,15 @@
     static {
         CREATE_INDEX = getBooleanProperty("create_index", false);
         CREATE_MD5 = getBooleanProperty("create_md5", false);
-        USE_ASYNC_IO = getBooleanProperty("use_async_io", false);
+        if (hasProperty("use_async_io")){
+            USE_ASYNC_IO = getBooleanProperty("use_async_io", false);
+            USE_ASYNC_IO_FOR_SAMTOOLS = USE_ASYNC_IO;
+            USE_ASYNC_IO_FOR_TRIBBLE = USE_ASYNC_IO;
+        } else {
+            USE_ASYNC_IO = false; //there's no good setting for it so we just use the default and deprecate the field to discourage users
```

Comment:
Done. Semantics is still kind of weird if someone specifies only `USE_ASYNC_IO_FOR_SAMTOOLS` and `USE_ASYNC_IO_FOR_TRIBBLE` and sets both to `true` (the `USE_ASYNC_IO` will remain `false`). But that can be lived with I guess.